packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -121,8 +121,11 @@ export class RootDataObject extends DataObject<{ InitialState: RootDataObjectPro
     private async createDataObject<T extends IFluidLoadable>(dataObjectClass: DataObjectClass<T>): Promise<T> {
         const factory = dataObjectClass.factory;
         const packagePath = [...this.context.packagePath, factory.type];
-        const router = await this.context.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<T>(router, "/");
+        const dataStore = await this.context.containerRuntime.createDataStore(packagePath);
+        const maybeIFluidHandle: IDataStore & FluidObject<IFluidHandle> = dataStore;
+        const entrypoint = maybeIFluidHandle.IFluidHandle?.get();
+        assert(entrypoint instanceof DataObject, "The data store's handle is not of the correct type!");
+        return entrypoint as unknown as T;
     }
```

Comment:
Not sure about a good way to validate this without `instanceOf`, since `DataObject` doesn't implement any interfaces exclusive to it that `FluidObject` can use to confirm that the object is of certain shape. Is it better to just drop the assert and trust that a function to initialize the entry point was provided to the data store runtime? Seems like only RootDataObjectFactory (which extends DataObjectFactory) creates data stores for this data object, and the changes there should guarantee that the entry point initialization function is provided.

Reply:
Removed the assert in the end